continuous-integration/code-generation/datamodel_frontend_generator.py

```diff
@@ -518,6 +532,7 @@ def _parse_internal_datamodel(current_class: Any) -> NodeData:
         snake_case=_to_snake_case(current_class.name),
         is_leaf=is_leaf,
         is_enum=is_enum_class,
+        is_array_like=is_leaf and current_class.is_array_like,
```

Comment:
is there a situation where `current_class.is_array_like` is true and `is_leaf` false?

Reply:
@knzivid Eskil is away for 3 weeks, do you think this is a blocker for merging this?

Reply:
No, not if is_array_like in C++ is set to false for non leaf nodes? I am not sure what _array_ here implies, so this suggestion could be wrong.

Reply:
Figured why this exists - this is equivalent to `hasattr(current_class, "is_array_like") and current_class.is_array_like`. The `is_leaf` check assumes the C++ implementation detail that `is_array_like` is only set for leaf nodes. I propose using hasattr or adding a comment about this implementation detail.

Reply:
added comment